feat(datepicker): vue3 style and keyboard navigation enhancement #95

iropolo opened the pull request:
# Vue3 Datepicker style and keyboard navigation enhancement

Vue2:  #95

arjundialpad commented:
Sorry, don't think I will be able to constructively contribute here.

tedgoas-dialpad commented:
@arjundialpad If you compare [the current buttons page](https://dialpad.design/components/buttons/) to [Josh's buttons](https://deploy-preview-98--dialtone.netlify.app/components/buttons/), you can visually compare the change in a browser ✌️

iropolo committed:
feat: datepicker refactor

francisrupert reviewed (changes requested):
Each button in Previous Month/Year row should be `muted` style.

![image](https://github.com/dialpad/dialtone/assets/1165933/7c715f4f-6c74-471b-a2eb-1c79449537c3)

iropolo committed:
feat: datepicker refactor

iropolo commented:
> Each button in Previous Month/Year row should be `muted` style.
> 

Right, seems like I forget to move more styles. Now I just directly use `d-btn--muted` class.

Updated in this vue3 PR on commit https://github.com/dialpad/dialtone/pull/98/commits/64ab02ba6f5c2ca76d2190850a3c96f8a2716b77

Updated also in vue2 PR #95 commit https://github.com/dialpad/dialtone/pull/95/commits/619731221b45bcc2933755c8d3844a57dec5f233

francisrupert commented on packages/dialtone-vue3/components/datepicker/modules/month-year-picker.vue:
```diff
@@ -1,73 +1,132 @@
+            :ref="el => { if (el) setDayRef(el) }"
+            size="xs"
+            importance="clear"
+            class="d-btn--icon-only d-datepicker__nav-btn d-btn--muted"
```
Should be able to do...
```
size="xs"
importance="clear"
kind="muted"
circle
class="d-datepicker__nav-btn"
```

iropolo replied:
👍🏼 
Added 
```
kind="muted"
circle
```
and clered class to `class="d-datepicker__nav-btn"` 

Updated in this vue3 PR on commit 
[03bcd03](https://github.com/dialpad/dialtone/pull/98/commits/03bcd033cff7daa9b6a0dc324e82e2fe6462778e)

PR #95 vue2 commit https://github.com/dialpad/dialtone/pull/95/commits/760a0f3eb3862521934b5ffb0d25c91d1c5c023d

iropolo committed:
feat: datepicker refactor
